### ISSUE TITLE
Unsupported api-version for armservicebus ListKeys() method

### ISSUE THREAD

**Issue by ajain-cldcvr** [Azure/azure-sdk-for-go#17520]
### Bug Report

Recently I upgrade `github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/servicebus/armservicebus` to  `v0.3.1`

I am using `ListKeys()` method from the `NamespacesClient` which is defined [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/servicebus/armservicebus/zz_generated_namespaces_client.go#L728)

But After the upgrade I started to receive error on `ListKeys()` api call and error snippet is as below 

```
{
	"error": {
		"code": "NoRegisteredProviderFound",
		"message": "No registered resource provider found for location 'centralus' and API version '2021-11-01' for type 'namespaces/authorizationrules'. The supported api-versions are 2014-09-01, 2015-08-01, 2017-04-01, 2018-01-01-preview, 2021-01-01-preview, 2021-06-01-preview. The supported locations are ''."
	}
}

```
I see that SDK is using an api-version which is not supported and hardcoded [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/servicebus/armservicebus/zz_generated_namespaces_client.go#L767)

For now to unblock myself I stopped using SDK for this particular function and shifted to [REST API](https://docs.microsoft.com/en-us/rest/api/servicebus/stable/namespaces-authorization-rules/list-keys)

**Comment by ArcturusZhang** [comment]
Hi @ajain-cldcvr sorry for the inconvenience, I am adding labels to loop the persons from the service side.

And in the future, we might add functionalities to allow you override the default api-versions
cc @jhendrixMSFT

**Comment by jsquire** [comment]
@ArcturusZhang:  The service team reached out today to understand what was being requested of them here.  After reading this over, I don't believe that there is any.  The inquiry appears to have been about a pre-release version of the library that has since been superseded by several GA releases.

Please review and if action is still needed, please clarify the ask on the service team.   Otherwise, please resolve or route as appropriate.

**Comment by Alancere** [comment]
NamespacesClient ListKeys is available ([live test](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/servicebus/armservicebus/namespace_live_test.go#L351) passed)

[Register resource provider](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-providers-and-types?wt.mc_id=knwlserapi_inproduct_azportal#register-resource-provider)
